contracts/Front/ShelteringTransfers.sol

```diff
@@ -67,28 +80,68 @@ contract ShelteringTransfers is Base {
     }
 
     function getDonor(bytes32 transferId) public view returns(address) {
-        (address donorId, ) = shelteringTransfersStore.getTransfer(transferId);
+        (address donorId,, ) = shelteringTransfersStore.getTransfer(transferId);
         return donorId;
     }
 
     function getTransferredBundle(bytes32 transferId) public view returns(bytes32) {
-        (, bytes32 bundleId) = shelteringTransfersStore.getTransfer(transferId);
+        (, bytes32 bundleId, ) = shelteringTransfersStore.getTransfer(transferId);
         return bundleId;
     }
 
     function transferIsInProgress(bytes32 transferId) public view returns(bool) {
-        (address donorId, ) = shelteringTransfersStore.getTransfer(transferId);
+        (address donorId,, ) = shelteringTransfersStore.getTransfer(transferId);
         return donorId != address(0x0);
     }
 
+    function canResolve(address resolverId, bytes32 transferId) public view returns (bool) {
+        (address donorId, bytes32 bundleId, ) = shelteringTransfersStore.getTransfer(transferId);
+
+        // solium-disable-next-line operator-whitespace
+        return donorId != address(0x0) &&
+        !sheltering.isSheltering(bundleId, resolverId) &&
+        resolverId == getTransferDesignatedShelterer(transferId);
+    }
+
+    function getTransferCreationTime(bytes32 transferId) public view returns (uint64) {
+        (,, uint64 creationTime) = shelteringTransfersStore.getTransfer(transferId);
+        return creationTime;
+    }
+
+    function getTransferDesignatedShelterer(bytes32 transferId) public view returns (address) {
```

Comment:
`getTransferDesignatedShelterer ` and `getDesignatedSheltererTier` are duplicates of methods in `Challenges`. Please extract superclass.

Reply:
getDesignatedSheltererTier - is ok!
But getTransferDesignatedShelterer uses the transfer creation time for round computation and transferId as dmpBaseHash.